src/AtomShell/process.jl

```diff
@@ -117,14 +117,32 @@ end
 
 export active
 
-active(shell::Electron) = process_running(shell.proc)
+Base.close(shell::Electron) = close(shell.sock)
+Base.isopen(shell::Electron) = process_running(shell.proc)
 
-quit(shell::Electron) = close(shell.sock)
+# TODO: Deprecate these functions in favor of methods above.
+active(shell::Electron) = isopen(shell)
+quit(shell::Electron) = close(shell)
 
 # Default process
 
-function shell(; debug = false)
-  global _shell
-  _shell ≠ nothing && active(_shell) && return _shell
-  _shell = init(debug = debug)
+_active_electron_process = Ref{Union{Electron, Nothing}}(nothing)
+
+"""
+    shell([debug=false])
+
+Get the currently active Electron shell instance (or activate one if none exists
+yet).
+
+NOTE: The `debug` keyword argument only takes effect if there is not an active
+`Electron` instance. If there *is* an active instance, this function returns
+that instance without regards to whether or not the `debug` setting matches.
```

Comment:
If you want to be super thorough, we could check if the requested debug settings match the existing debug settings, and if not, print a warning or something?